packages/hardhat-core/src/internal/core/errors-list.ts

```diff
@@ -237,6 +237,21 @@ Either try using a new directory name, or remove the conflicting files.`,
         "Hardhat attempted to convert the input value to a BigInt, but no known conversion method was applicable to the given value.",
       shouldBeReported: false,
     },
+    CORRUPTED_LOCKFILE: {
+      number: 18,
+      message: `You installed Hardhat with a corrupted lockfile due to the NPM bug #4828.
+
+Please delete your node_modules, package-lock.json, reinstall your project, and try again.`,
+      title: "Corrupted lockfile",
+      description: `Some versions of NPM are affected [by a bug](https://github.com/npm/cli/issues/4828) that leads to corrupt lockfiles being generated.
+
+This bug can only affect you if you are installing your project without a lockfile, but with an existing node_modules.
```

Comment:
Maybe I'm missing something here, but I think this is misleading, because this is what the person that generated the corrupted lockfile did, not what the person finding the bug did.

So I guess the message should be something like: "This bug can happen if one of your teammates generated the lockfile using a different operating system or architecture."

Reply:
great point!